outlines/models/vllm.py

```diff
@@ -22,6 +23,23 @@ def __init__(self, model: "LLM"):
         self.model = model
         self.lora_request = None
 
+        self.tokenizer = self._get_tokenizer()
+
+    def _get_tokenizer(self):
+        if hasattr(self.model, "get_tokenizer"):
+            tokenizer = self.model.get_tokenizer()
```

Comment:
Heads up, for the `AyncLLMEngine` (as shown in [the `outlines` vLLM example](https://github.com/outlines-dev/outlines/blob/main/outlines/serve/serve.py#L126)), this will return a coroutine: https://github.com/vllm-project/vllm/blob/main/vllm/engine/async_llm_engine.py#L506 .

I'm trying to figure out the best path forward because I'd love to use this with my vLLM-based service, but it seems like this work is [part of something bigger](https://github.com/outlines-dev/outlines/issues/1054) so I don't want to dive in and start propagating `async` through this code without checking in with you first.  Happy to contribute, but could use a little guidance on the strategy 😁

Reply:
It's uncertain whether we will move towards async for `outlines.generate`, but it has been proposed https://github.com/outlines-dev/outlines/issues/655

Bare in mind that `outlines.serve` already has a [vllm server integration](https://github.com/outlines-dev/outlines/blob/main/docs/reference/serve/vllm.md#L4) and vice versa, [vllm has an `outlines.processors` integration in progress](https://github.com/vllm-project/vllm/pull/6715)

Does `outlines.serve` or vLLM's outlines integration satisfy your needs, or were you thinking of something different?

Reply:
Ahhh, thank you for the sanity check!  After re-reviewing the `outlines.serve` code, I realized I didn't go deep enough and needed to pass my `engine.engine` (engines all the way down 🐢) to get all the way to the `vllm.LLM`.  Thanks again for the pointers!

Reply:
No problem! Bare in mind that after our next major (**because** of this PR), the **tokenizer**, not the engine will be passed to the processor. `serve.py` has a PR to reflect this behavior https://github.com/outlines-dev/outlines/pull/1061/files#diff-535a1da5f8addb89d07782185c32b54f85189b25786d1c9b7cbd002b55939e16R74